eventsources/eventing.go

```diff
@@ -422,6 +445,23 @@ func (e *EventSourceAdaptor) run(ctx context.Context, servers map[apicommon.Even
 						if err != nil {
 							return err
 						}
+
+						if filter, ok := filters[getFilterKey(s.GetEventSourceType(), s.GetEventName())]; ok {
```

Comment:
Move the whole section to line 428, the filtering should happen before everything else.